llvm/lib/ExecutionEngine/JITLink/EHFrameSupport.cpp

```diff
@@ -61,6 +61,14 @@ Error DWARFRecordSectionSplitter::operator()(LinkGraph &G) {
       return Err;
   }
 
+  // If the debug_line section is being split, add an anonymous symbol so that
+  // the blocks don't get dropped
+  if (SectionName == "__DWARF,__debug_line") {
+    for (auto *B : Section->blocks()) {
+      G.addAnonymousSymbol(*B, 0, B->getSize(), false, true);
+    }
+  }
+
```

Comment:
This should be outside DWARFRecordSectionSplitter entirely.

Something like this would work for now:
```
LinkGraphPassFunction createDebugLineSplitterPass(StringRef DebugLineSectionName) {
  return [DebugLineSectionName](LinkGraph &G) {
    if (auto Err = DWARFRecordSectionSplitter(DebugLineSectionName)(G))
      return Err;
    if (auto *DLSec = G.findSectionByName(DebugLineSectionName)) {
      for (auto *B : DLSec->blocks())
        G.addAnonymousSymbol(*B, 0, B->getSize(), false, true);
    }
    return Error::success();
  };
}
```